### ISSUE TITLE
Little issues setting up flow

### ISSUE THREAD

**Issue by bsansouci** [facebook/react-native#3007]
Hey guys,

I just tried running `flow` on my AwesomeProject and it told me that my version (0.16.0) was different from the one it he .flowconfig file (0.14.0). So I changed it to 0.16.0 and got a bunch (160) of errors. I also tried installing v0.14.0 and get similar errors. They all look exactly the same for different files:
```
/Users/sansouci/Desktop/AwesomeProject/node_modules/react/lib/ReactDOMTextComponent.js: ReactDOMTextComponent
Duplicate module provider
/Users/sansouci/Desktop/AwesomeProject/node_modules/react-native/node_modules/react-tools/src/renderers/dom/shared/ReactDOMTextComponent.js: current provider

/Users/sansouci/Desktop/AwesomeProject/node_modules/react/lib/ReactDOMSelection.js: ReactDOMSelection
Duplicate module provider
/Users/sansouci/Desktop/AwesomeProject/node_modules/react-native/node_modules/react-tools/src/renderers/dom/client/ReactDOMSelection.js: current provider
```

Do you guys have any idea why this is happening?

**Comment by ide** [comment]
Flow is not 100% backwards-compatible since it is under heavy development. Flow 0.16 support is coming with the next release of React Native.  Till then you could get Flow 0.14 from https://www.npmjs.com/package/flow-bin and try that.

**Comment by bsansouci** [comment]
I did try flow 0.14.0 and had similar issues. Let me try again from the link you gave me.

**Comment by bsansouci** [comment]
It seems like it's just download the binary from github.com/flow, which I already tried (I've tried 0.14 and 0.16)

**Comment by ide** [comment]
Ah ok. Too bad. Sorry I don't have a better suggestion.

**Comment by ku-kueihsi** [comment]
0.16.0 is not working. I tried 0.14.0 osx binary on github. That works and there is no errors for AwesomeProject.

**Comment by joshhornby** [comment]
I seem to be having some issues with version `0.18.1`

I am seeing errors such as:

```
node_modules/react-native/Libraries/Animated/src/AnimatedImplementation.js:17
 17: var React = require('React');
                 ^^^^^^^^^^^^^^^^ react.js. Required module not found

node_modules/react-native/Libraries/Animated/src/Interpolation.js:205
205:   var outputRanges = outputRange[0].match(stringShapeRegex).map(() => []);
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ call of method `match`
205:   var outputRanges = outputRange[0].match(stringShapeRegex).map(() => []);
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ call of method `map`. Method cannot be called on possibly null value
205:   var outputRanges = outputRange[0].match(stringShapeRegex).map(() => []);
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ null
```

Think docs might need updating in terms of setting up Flow and React Native

**Comment by brentvatne** [comment]
cc @gabelevi

**Comment by jwhitley** [comment]
I've seen errors similar to the above previously, and just now I inadvertently updated flow to `0.21.0` when I did a `brew upgrade`.  I was a bit flabbergasted that there appeared to be parts of RN which didn't pass `flow check`.  But the actual problem was a silent mismatch between my `flow` and `.flowconfig` and my app's dependencies' required version of `flow` (i.e. `react-native`)

The documentation and workflow (esp. around updating the RN version) here definitely need work.  I feel like I need to add some ugly script layer in my app's dev workflow that looks at `node_modules/react-native/.flowconfig` and compares its version to my app's own `.flowconfig` version.  And then breaks the world if they don't agree.  One could argue that `flow` should do this check itself, breaking if a app depends on a code in a subdirectory with an incompatible `.flowconfig`, but that probably introduces another hornet's nest of issues.

**Comment by corbt** [comment]
Flow support is working with React Native with the latest versions of RN and Flow. If you use the default generated `.flowconfig` file that is created for you when you run `react-native init` or `react-native upgrade` you should be fine.
